src/lambda/wizards/samDeployWizard.ts

```diff
@@ -635,13 +685,26 @@ export class SamDeployWizard extends MultiStepWizard<SamDeployWizardResponse> {
 
     private readonly S3_BUCKET: WizardStep = async step => {
         const response = await this.context.promptUserForS3Bucket(step, this.response.region, this.response.s3Bucket)
-
-        this.response.s3Bucket = response
-
+        
         if (!response) {
             return WIZARD_GOBACK
         }
 
+        if(response === CREATE_NEW_BUCKET) {
```

Comment:
This is an improvement. The use of `this.newBucketCalled` still feels clunky but less fragile at least.